src/Compilers/CSharp/Test/Syntax/IncrementalParsing/IncrementalParsingTests.cs

```diff
@@ -3666,6 +3666,33 @@ enum VirtualKey
             WalkTreeAndVerify(tree.GetCompilationUnitRoot(), fullTree.GetCompilationUnitRoot());
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/76439")]
+        public void InKeywordInsideAForBlock()
+        {
+            var source = """
+                void Main()
+                {
+                    for (int i = 0; i < n; i++)
+                    {
+                    }
+                }
+                """;
+            var tree = SyntaxFactory.ParseSyntaxTree(source);
+            var text = tree.GetText();
+
+            var position1 = source.IndexOf("i =") + 1;
+            var position2 = source.IndexOf("i <") + 1;
+            var position3 = source.IndexOf("i++") + 1;
+
+            text = text.WithChanges(
+                new TextChange(new TextSpan(position1, 0), "n"),
+                new TextChange(new TextSpan(position2, 0), "n"),
+                new TextChange(new TextSpan(position3, 0), "n"));
+            tree = tree.WithChangedText(text);
```

Comment:
It would make the test slightly easier to grasp if we asserted the contents of the resulting text here.

Reply:
done.